rest-api/src/main/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/CaseDefinitionController.java

```diff
@@ -142,4 +154,22 @@ public CaseTypeVersionInformation dataCaseTypeVersionGet(@ApiParam(value = "Case
                                                                  String id) {
         return caseTypeService.findVersionInfoByCaseTypeId(id).orElseThrow(() -> new NotFoundException(id));
     }
+
+
+    @GetMapping(value = "/data/jurisdictions/case-type-ids")
+    @ApiOperation(value = "Get case-type-ids by jurisdiction's id", response = String.class, responseContainer = "List")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "List of cases ids")
+    })
+    public List<String> findCasesByJurisdictions(
+        @ApiParam(value = "list of jurisdiction references") @RequestParam("ids") Optional<List<String>> idsOptional) {
+
+        LOG.debug("received find jurisdictions request with ids: {}", idsOptional);
+        final var  jurisdictions = idsOptional.map(ids -> jurisdictionService.getAll(ids))
+            .orElseGet(jurisdictionService::getAll);
+
+        return jurisdictions.stream().flatMap(jurisdiction -> jurisdiction.getCaseTypes().stream().map(
+            caseTypeLite -> caseTypeLite.getId())
+        ).collect(Collectors.toList());
```

Comment:
Potential improvement: Since we are only after the references as part of this call, it may be worth introducing a new query to `CaseTypeRepository` `findAllCaseTypeIds(List<String> jurisdictionIds)` which queries something like `select <case type references> from case_type where the case type's jurisdiction in (<provided ids>)`. Then this logic wouldn't need to call `jurisdictionService` and get a list which it then has to process; instead this could become: 

```suggestion
        return idsOptional.map(caseTypeService::findAllCaseTypeIds)
            .orElseGet(caseTypeService::findAllCaseTypeIds);
```

Reply:
Yes I think this is a very good point. It would be beneficial  adding these new service and only retrieve  "case reference" from the caseType  for some  jurisdictions in the caseTypeService itself  instead of retrieving them from the  jurisdictionService.